kroxylicious-integration-tests/src/test/java/io/kroxylicious/proxy/MetricsIT.java

```diff
@@ -152,6 +161,170 @@ void prometheusMetricsWithCommonTags(KafkaCluster cluster) {
+                .endEndpoints()
+                .endManagement();
+
+        try (var tester = kroxyliciousTester(config);
+                var managementClient = tester.getManagementClient()) {
+
+            var metricsList = managementClient.scrapeMetrics();
+
+            assertThat(checkMetricsExistOrNot(metricsList, "kroxylicious_downstream_connections_total", null)).isEqualTo(false);
+            assertThat(checkMetricsExistOrNot(metricsList, "kroxylicious_upstream_connections_total", null)).isEqualTo(false);
+
+            var producer = tester.producer();
+            producer.send(new ProducerRecord<>(topic.name(), "my-key", "hello-world")).get();
+
+            // updated metrics after some message were produced
+            var updatedMetricsList = managementClient.scrapeMetrics();
+
+            assertThat(checkMetricsExistOrNot(updatedMetricsList, "kroxylicious_downstream_connections_total", null)).isEqualTo(true);
+
+            assertThat(getMetricsValue(updatedMetricsList, "kroxylicious_downstream_connections_total", null)).isGreaterThan(0);
+            assertThat(getMetricsValue(updatedMetricsList, "kroxylicious_upstream_connections_total", null)).isGreaterThan(0);
+        }
+    }
+
+    @Test
+    void shouldIncrementPayloadSizeBytesMetricsOnProduceRequest(KafkaCluster cluster, Topic topic) throws ExecutionException, InterruptedException {
+        var config = proxy(cluster)
+                .addToFilterDefinitions()
+                .addToDefaultFilters()
+                .withNewManagement()
+                .withNewEndpoints()
+                .withNewPrometheus()
+                .endPrometheus()
+                .endEndpoints()
+                .endManagement();
+
+        try (var tester = kroxyliciousTester(config);
+                var managementClient = tester.getManagementClient();
+                var producer = tester.producer();) {
+
+            var metricList = managementClient.scrapeMetrics();
+
+            assertThat(checkMetricsExistOrNot(metricList, "kroxylicious_payload_size_bytes_count", ApiKeys.PRODUCE)).isEqualTo(false);
+            assertThat(checkMetricsExistOrNot(metricList, "kroxylicious_payload_size_bytes_sum", ApiKeys.PRODUCE)).isEqualTo(false);
+
+            producer.send(new ProducerRecord<>(topic.name(), "my-key", "hello-world")).get();
+            producer.send(new ProducerRecord<>(topic.name(), "my-key", "hello-world")).get();
+
+            // updated metrics after some message were produced
+            var updatedMetricList = managementClient.scrapeMetrics();
+
+            assertThat(checkMetricsExistOrNot(updatedMetricList, "kroxylicious_payload_size_bytes_count", ApiKeys.PRODUCE)).isEqualTo(true);
+            assertThat(getMetricsValue(updatedMetricList, "kroxylicious_payload_size_bytes_count", ApiKeys.PRODUCE)).isGreaterThan(0);
+        }
+    }
+
+    boolean checkMetricsExistOrNot(List<SimpleMetric> metricList, String metricsName, ApiKeys apiKey) {
```

Comment:
Split this into two methods.

`checkMetricExists` and `checkMetricTaggedWithApIKey` or something.

It makes it much easier to read what we are trying to show in the test cases. On naming I think the `OrNot` suffix is just making the name longer and clunky. The `OrNot` part is implicit in the fact it has a boolean return value.